python/turbodbc_test/test_select_arrow.py

```diff
@@ -54,6 +56,8 @@ def test_arrow_empty_column(dsn, configuration):
             assert isinstance(result, pa.Table)
             assert result.num_columns == 1
             assert result.num_rows == 0
+            gc.collect()
```

Comment:
I'd prefer the refcount check in a separate test with a clear name so that this test has a single intent.